service/witness/witness/witness.go

```diff
@@ -204,12 +197,43 @@ func (w *Witness) RescheduleBlockWitness() {
 	if time.Now().After(nextBlockWitness.UpdatedAt.Add(UnprovedBlockWitnessTimeout)) {
 		err := w.blockWitnessModel.UpdateBlockWitnessStatus(nextBlockWitness, blockwitness.StatusPublished)
 		if err != nil {
-			logx.Errorf("update unproved block status error, err: %v", err)
+			logx.Errorf("update unproved block status error, err: %s", err.Error())
 			return
 		}
 	}
 }
 
+func (w *Witness) getNextWitnessToCheck() (int64, error) {
+	latestProof, err := w.proofModel.GetLatestProof()
+	if err != nil && err != types.DbErrNotFound {
+		logx.Errorf("failed to get latest proof, err: %s", err.Error())
+		return 0, err
+	}
+
+	if err == types.DbErrNotFound {
+		return 1, nil
+	}
+
+	latestConfirmedProof, err := w.proofModel.GetLatestConfirmedProof()
+	if err != nil && err != types.DbErrNotFound {
+		logx.Errorf("failed to get latest confirmed proof, err: %s", err.Error())
+		return 0, err
+	}
+
+	var startToCheck, endToCheck int64 = 1, latestProof.BlockNumber
+	if err != types.DbErrNotFound {
+		startToCheck = latestConfirmedProof.BlockNumber + 1
+	}
+
+	for blockHeight := startToCheck; blockHeight <= endToCheck; blockHeight++ {
```

Comment:
Can we replace `blockHeight <= endToCheck` to `blockHeight < endToCheck`, because endToCheck proof must exist